akka-cluster-metrics/src/main/scala/akka/cluster/metrics/ClusterMetricsRouting.scala

```diff
@@ -0,0 +1,509 @@
+ *
+ * @param paths string representation of the actor paths of the routees, messages are
+ *   sent with [[akka.actor.ActorSelection]] to these paths
+ *
+ * @param routerDispatcher dispatcher to use for the router head actor, which handles
+ *   router management messages
+ */
+@SerialVersionUID(1L)
+final case class AdaptiveLoadBalancingGroup(
+  metricsSelector: MetricsSelector = MixMetricsSelector,
+  paths: immutable.Iterable[String] = Nil,
+  override val routerDispatcher: String = Dispatchers.DefaultDispatcherId)
+  extends Group {
+
+  def this(config: Config, dynamicAccess: DynamicAccess) =
+    this(metricsSelector = MetricsSelector.fromConfig(config, dynamicAccess),
+      paths = immutableSeq(config.getStringList("routees.paths")))
+
+  /**
+   * Java API
+   * @param metricsSelector decides what probability to use for selecting a routee, based
+   *   on remaining capacity as indicated by the node metrics
+   * @param routeesPaths string representation of the actor paths of the routees, messages are
+   *   sent with [[akka.actor.ActorSelection]] to these paths
+   */
+  def this(metricsSelector: MetricsSelector,
+           routeesPaths: java.lang.Iterable[String]) = this(paths = immutableSeq(routeesPaths))
+
+  override def createRouter(system: ActorSystem): Router =
+    new Router(AdaptiveLoadBalancingRoutingLogic(system, metricsSelector))
+
+  override def routingLogicController(routingLogic: RoutingLogic): Option[Props] =
+    Some(Props(classOf[AdaptiveLoadBalancingMetricsListener],
+      routingLogic.asInstanceOf[AdaptiveLoadBalancingRoutingLogic]))
+
+  /**
+   * Setting the dispatcher to be used for the router head actor, which handles
+   * router management messages
+   */
+  def withDispatcher(dispatcherId: String): AdaptiveLoadBalancingGroup = copy(routerDispatcher = dispatcherId)
+
+}
+
+/**
+ * MetricsSelector that uses the heap metrics.
+ * Low heap capacity => small weight.
+ */
+@SerialVersionUID(1L)
+case object HeapMetricsSelector extends CapacityMetricsSelector {
+  import akka.cluster.metrics.StandardMetrics.HeapMemory
+  /**
+   * Java API: get the singleton instance
+   */
+  def getInstance = this
+
+  override def capacity(nodeMetrics: Set[NodeMetrics]): Map[Address, Double] = {
+    nodeMetrics.collect {
+      case HeapMemory(address, _, used, committed, max) ⇒
+        val capacity = max match {
+          case None    ⇒ (committed - used).toDouble / committed
+          case Some(m) ⇒ (m - used).toDouble / m
+        }
+        (address, capacity)
+    }.toMap
+  }
+}
+
+/**
+ * MetricsSelector that uses the combined CPU metrics.
+ * Combined CPU is sum of User + Sys + Nice + Wait, in percentage.
+ * Low cpu capacity => small weight.
+ */
+@SerialVersionUID(1L)
+case object CpuMetricsSelector extends CapacityMetricsSelector {
+  import akka.cluster.metrics.StandardMetrics.Cpu
+  /**
+   * Java API: get the singleton instance
+   */
+  def getInstance = this
+
+  // TODO introduce cpu stolen time
```

Comment:
cpuCombined and cpuStolen add up to more than 1.0, its 2 separate dimensions.
I could not come up with proper logic / weight factor